### PR TITLE
Warn users about bad equiprobable angles in S(a,b)

### PR DESCRIPTION
**Pull request by HunterBelanger**
# Description

A while back, I made some fixes to the TSL processing methodology (see #2290). Back then, I mentioned adding a warning for discrete scattering cosines being outside of [-1,1] (as this could happen with not-that-old versions of NJOY). I forgot to make the second PR for that until now.

I decided to make this a warning and not an error, simply because there are so many ACE files out there with this bug, particularly those distributed with MCNP for ENDF/B-VII.1. Users might want to run with that data, to make comparisons against different libraries or codes.

# Checklist

- [X] I have performed a self-review of my own code
- [ ] <s>I have run [clang-format](https://docs.openmc.org/en/latest/devguide/styleguide.html#automatic-formatting) on any C++ source files (if applicable)</s>
- [X] I have followed the [style guidelines](https://docs.openmc.org/en/latest/devguide/styleguide.html#python) for Python source files (if applicable)
- [ ] <s>I have made corresponding changes to the documentation (if applicable)</s>
- [ ] <s>I have added tests that prove my fix is effective or that my feature works (if applicable)</s>

### CODE REVIEW

File: openmc/data/thermal.py [hunk 1]
```diff
@@ -676,6 +676,15 @@ def from_ace(cls, ace_or_filename, name=None):
                     # here, because they are equiprobable, so the order
                     # doesn't matter.
                     mu.sort()
+
+                    # Older versions of NJOY had a bug, and the discreet
```
Comment (gridley) [inline openmc/data/thermal.py:680]:
```suggestion
                    # Older versions of NJOY had a bug, and the discrete
```

File: openmc/data/thermal.py [hunk 2]
```diff
@@ -676,6 +676,15 @@ def from_ace(cls, ace_or_filename, name=None):
...
+                    # scattering angles could sometimes be less than -1 or
+                    # greater than 1. We check for this here, and warn users.
+                    if mu[0] < -1. or mu[-1] > 1.:
```
Comment (gridley) [inline openmc/data/thermal.py:683]: IMO we should just go ahead and check if `np.any(np.abs(mu) > 1.0)` and warn if so, out of an abundance of caution.

**Review by gridley — CHANGES_REQUESTED** [review]

File: openmc/data/thermal.py (hunk 2 above)
Comment (HunterBelanger) [reply openmc/data/thermal.py:683]: This was certainly a problem before, as I was finding many files with discrete angles which were not necessarily in ascending order. That was likely causing bad approximations to be made with the angle-smearing operation to "smooth" out the distribution, so I added a sort on mu in the previous PR. Now that mu is guaranteed to be in ascending order, I think just checking the first and last elements should suffice ?

**Review by paulromano — APPROVED** [review]
LGTM. Thanks @HunterBelanger!

File: openmc/data/thermal.py (hunk 2 above)
Comment (gridley) [reply openmc/data/thermal.py:683]: OK, makes sense!

**Comment by paulromano** [comment]
@gridley If you're satisfied with the changes, can you approve this? Thanks!

**Pull request merged**